### PR TITLE
Added mxint_matrix_cat.sv

### PR DESCRIPTION
**Pull request by omaralkhatib03**
This PR intends to implement and integrate a matrix concatenation module into MASE. The module implements the equivalent of `torch.cat([t1, t2], dim=-1)` in HW. 

The module assumes that the streaming is always happening in the last
direciton, i.e the concatenating direction.

E.g to concatenate the following 2 x 2 matricies:

      1   2       5   6       -->   1   2   5   6
      3   4       7   8       -->   3   4   7   8

The data is expected to be streamed left to right in blocks. So for a block
size of 1, this module expects 1 2 3 4, and 5 6 7 8 on each input interface.
The output would be 1 2 5 6 3 4 7 8, hence concatenating the two matricies.

The limitation of this module is that it requires the PARALLELISM parameters
of the input and output interfaces to be the same. This mainly a limitation
of mxint_cast. Can later remove this by adding a flow controller on the
input of the module. The flow controller would ensure that the concatenation
internally happens using uniform PARALLELISM. This is not a trivial

## Status

1. The module is untested at the minute. However, I know we are making changes to the module interfaces/parameters. Please provide the necessary feedback regarding the interface when the mx_int_layer is finalised. I will continue working on the PR (writing the TB now) until further notice.

### CODE REVIEW

File: src/mase_components/linear_layers/mxint_operators/test/config.gtkw
Comment (luigirinaldi) [inline src/mase_components/linear_layers/mxint_operators/test/config.gtkw:1]: Why is this here??

File: src/mase_components/linear_layers/mxint_operators/rtl/mxint_matrix_cat.sv [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    always_comb begin
+      for (int i = 0; i < BLOCK_SIZE; i++) begin
+        mdata_in_1_c[i] = mdata_in_1[i];
+      end
+      shift_in_1      = data_in_1_valid;
+      data_in_1_ready = fifo_1_full;
+      edata_in_1_c    = edata_in_1;
+    end
+  end else begin : cast_in_1_gen
+    mxint_cast #(
```
Comment (luigirinaldi) [inline src/mase_components/linear_layers/mxint_operators/rtl/mxint_matrix_cat.sv:197]: this is not what cast does you dufus

Comment (luigirinaldi) [reply src/mase_components/linear_layers/mxint_operators/rtl/mxint_matrix_cat.sv:197]: why would you think it performs a cast?

File: src/mase_components/linear_layers/mxint_operators/test/config.gtkw
Comment (omaralkhatib03) [reply src/mase_components/linear_layers/mxint_operators/test/config.gtkw:1]: removed

**Review by luigirinaldi — APPROVED** [review]
lgtm
